plugins/database/hana/hana_test.go

```diff
@@ -49,38 +55,145 @@ func TestHANA_CreateUser(t *testing.T) {
 	if err == nil {
 		t.Fatal("Expected error when no creation statement is provided")
 	}
 
-	statements := dbplugin.Statements{
-		Creation: []string{testHANARole},
+	// Add a statement command
+	req.Statements.Commands = []string{testHANARole}
+
+	userResp, err = db.NewUser(context.Background(), req)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	if err = testCredsExist(t, connURL, userResp.Username, password); err != nil {
+		t.Fatalf("Could not connect with new credentials: %s", err)
+	}
+}
+
+func TestHANA_UpdateUser(t *testing.T) {
+	if os.Getenv("HANA_URL") == "" || os.Getenv("VAULT_ACC") != "1" {
+		t.SkipNow()
+	}
+	connURL := os.Getenv("HANA_URL")
+
+	connectionDetails := map[string]interface{}{
+		"connection_url": connURL,
+	}
+
+	initReq := newdbplugin.InitializeRequest{
+		Config:           connectionDetails,
+		VerifyConnection: true,
 	}
 
-	username, password, err := db.CreateUser(context.Background(), statements, usernameConfig, time.Now().Add(time.Hour))
+	db := new()
+	_, err := db.Initialize(context.Background(), initReq)
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
-	if err = testCredsExist(t, connURL, username, password); err != nil {
+	password, err := credsutil.RandomAlphaNumeric(32, true)
+	if err != nil {
+		t.Fatalf("failed to generate password: %s", err)
+	}
+	password = strings.Replace(password, "-", "_", -1)
+
+	newReq := newdbplugin.NewUserRequest{
+		UsernameConfig: newdbplugin.UsernameMetadata{
+			DisplayName: "test-test",
+			RoleName:    "test-test",
+		},
+		Password: password,
+		Statements: newdbplugin.Statements{
+			Commands: []string{testHANARole},
+		},
+		Expiration: time.Now().Add(time.Hour),
+	}
+
+	// Test default revoke statements
+	userResp, err := db.NewUser(context.Background(), newReq)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	if err = testCredsExist(t, connURL, userResp.Username, password); err != nil {
 		t.Fatalf("Could not connect with new credentials: %s", err)
 	}
+
+	newPassword, err := credsutil.RandomAlphaNumeric(32, true)
+	if err != nil {
+		t.Fatalf("failed to generate password: %s", err)
+	}
+	newPassword = strings.Replace(newPassword, "-", "_", -1)
+
+	// Change Password
+	updateReq := newdbplugin.UpdateUserRequest{
+		Username: userResp.Username,
+		Password: &newdbplugin.ChangePassword{
+			NewPassword: newPassword,
+		},
+	}
+
+	_, err = db.UpdateUser(context.Background(), updateReq)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	if err := testCredsExist(t, connURL, userResp.Username, newPassword); err == nil {
+		t.Fatal("Credentials were not changed")
+	}
+
+	// Test custom update statement
+	userResp, err = db.NewUser(context.Background(), newReq)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	if err = testCredsExist(t, connURL, userResp.Username, password); err != nil {
+		t.Fatalf("Could not connect with new credentials: %s", err)
+	}
+
+	updateReq.Password.Statements.Commands = []string{testHANAUpdate}
```

Comment:
There's a lot of shared state in these tests. Can these be made into table tests instead? MSSQL has some decent examples: https://github.com/hashicorp/vault/blob/master/plugins/database/mssql/mssql_test.go